### PR TITLE
To Do list: Interactive list

### PR DESCRIPTION
**Pull request by jcmunav63**
**Description**
In this project, you will add some functionality to your application to make it interactive. The user will also be able to mark task completion by selecting the corresponding checkbox (or undo it by unchecking the checkbox). The updated tasks list will be stored in local storage.

**General requirements**
- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used the correct [GitHub Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

**HTML/CSS requirements**
- Follow our list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).

**JavaScript requirements**
- Follow our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

**Project requirements**
- Add a new JavaScript file and import it as a module:
   - it will contain methods related to the status updates (completed: true/false).
- Add event listener to the checkbox (change).
- Update items object's value for completed key upon user actions.
Implement a function for the "Clear all completed" button (use filter() method).
- Store the updated array of items in local storage, so the user gets the correct list of values after the page reloads (which means that any changes made to the list should be preserved).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,10 +145,7 @@ To run tests, run the following command:
```
Comment (cynthiainga) [inline README.md:147]: - [ ] Please kindly fix the MIT broken link in your README file, having broken links in the README file doesn’t look very professional 😎

File: src/completed.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+function deleteCompletedTasks(tasksLocal) {
+  const taskCheckbox = document.querySelectorAll('.checkbox');
+  taskCheckbox.forEach((cb) => {
+    if (cb.checked) {
+      const li = cb.parentNode;
+      li.remove();
+    }
+  });
+  tasksLocal = tasksLocal.filter((tr) => tr.completed !== true);
+  // tasksLocal = tasksLocal.filter(!tasksLocal.completed);
```
Comment (cynthiainga) [inline src/completed.js]: - [ ] Please kindly delete this commented line of code to have a clean codebase, it is very important to have clean code 😉 😉

File: src/index.js [hunk 3]
```diff
@@ -46,32 +47,23 @@ const displayTaskElement = (task) => {
...
       const delBtn = parent.getElementsByClassName('delete-icon')[0];
+      const taskList = document.getElementById('task-list');
       if (delBtn.classList.contains('hide')) {
         delBtn.classList.remove('hide');
-        activateDeleteListener(delBtn, parent);
+        delBtn.addEventListener('click', () => {
+          deleteTaskElement(tasksLocal, delBtn);
+          loadTasksFromLocalStorage();
+          taskList.innerHTML = '';
+          /* eslint-disable no-use-before-define */
```
Comment (cynthiainga) [inline src/index.js]: - [ ] It's not a good approach to disable linters, fixing linters help us to write a clean and strong codebase 🙃 🙃

File: src/index.js [hunk 4]
```diff
@@ -86,42 +78,75 @@ function activateCheckboxListeners() {
...
+        updateTaskStatus(status, taskIndex, tasksLocal);
       } else {
+        status = false;
         taskInput.classList.remove('completed-task');
+        updateTaskStatus(status, taskIndex, tasksLocal);
       }
+      loadTasksFromLocalStorage();
+      const taskList = document.getElementById('task-list');
+      taskList.innerHTML = '';
+      /* eslint-disable no-use-before-define */
```
Comment (cynthiainga) [inline src/index.js]: - [ ] It's not a good approach to disable linters, fixing linters help us to write a clean and strong codebase 🙃 🙃

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @jcmunav63, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] The `Clear all completed` button works as expected ✔️ 
- [ ] If the box of the task is checked the status of the task change to true ✔️ 
- [ ] No linters errors ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BertrandConxy — APPROVED** [review]
Hi @jcmunav63 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💪 
- No linter errors
- project requirements are met
- project works fine
- good PR description
- correct flow is used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**